ocaml/lambda/switch.ml

```diff
@@ -660,17 +660,25 @@ let rec pkey chan  = function
           and right = {s with cases=right} in
 
           if i=1 && (lim+ctx.off)=1 && get_low cases 0+ctx.off=0 then
-            Arg.make_if
-              ctx.arg
-              (c_test ctx right) (c_test ctx left)
+            (* CR mshinwell (with trefis): This is related to PR#8672.
+               That PR seems to incorrectly make the assumption that being
+               here means that the scrutinee has range {0, 1}. *)
```

Comment:
I don't think that assumption was incorrect before this change to the semantics of Lifthenelse! See the discussion on #8672, particularly [this bit](https://github.com/ocaml/ocaml/pull/8672#discussion_r285206680).